frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/storage-tab-state-update.ts

```diff
@@ -16,13 +16,28 @@ import { DiskWrapper } from '../../../../k8s/wrapper/vm/disk-wrapper';
 
 export const prefillInitialDiskUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
   const state = getState();
-  if (!hasVmSettingsChanged(prevState, state, id, VMSettingsField.PROVISION_SOURCE_TYPE)) {
+  const baseDiskImageChanged = hasVmSettingsChanged(
+    prevState,
+    state,
+    id,
+    VMSettingsField.OPERATING_SYSTEM,
+    VMSettingsField.FLAVOR,
+    VMSettingsField.WORKLOAD_PROFILE,
+    VMSettingsField.CLONE_COMMON_BASE_DISK_IMAGE,
+  );
+
+  if (
+    !hasVmSettingsChanged(prevState, state, id, VMSettingsField.PROVISION_SOURCE_TYPE) &&
```

Comment:
we can use it here as well and maybe unify it in one call?

Reply:
done